GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -759,6 +781,17 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
             return result;
         }
 
+        private FileSystemTaskResult TryRemoveModifiedPath(string virtualPath, bool isFolder)
+        {
+            if (!this.modifiedPaths.TryRemove(virtualPath, isFolder, out bool isRetryable))
+            {
+                return isRetryable ? FileSystemTaskResult.RetryableError : FileSystemTaskResult.FatalError;
+            }
+
+            this.InvalidateGitStatusCache();

```

Comment:
Why did you add this line here? And how did you know you needed it?

In general, as we continue to make changes in this code, what are the principles we should keep in mind about when to invalidate the status cache? cc @jamill

Reply:
This is here because I was following the `TryAddModifiedPath` method and wanted to make sure the status cache would stay correct.  I will think this over more to determine if it is really needed and any input from @jamill would be great as well.

Reply:
We need to invalidate the git status cache when we observe / record operations that possibly affect the results of a Git status call. In this case, we are processing an event that indicates a file / directory on disk has been mutated, so this is appropriate.